src/main/java/jenkins/plugin/randomjobbuilder/GeneratorController.java

```diff
@@ -0,0 +1,365 @@
+package jenkins.plugin.randomjobbuilder;
+
+import com.google.common.collect.Sets;
+import hudson.Extension;
+import hudson.model.Executor;
+import hudson.model.Job;
+import hudson.model.PeriodicWork;
+import hudson.model.ReconfigurableDescribable;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.model.listeners.RunListener;
+import hudson.security.ACL;
+import jenkins.model.Jenkins;
+import org.acegisecurity.context.SecurityContext;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Controls load generation for an input set of runs, by watching a registered set of LoadGenerators
+ * And using listeners to add/remove tasks as desired
+ * Registers run for lookup, so we know which ones to kill if we stop load suddenly...
+ *   or in which cases we need to start new jobs to maintain load level
+ *
+ *   And triggers new builds when needed
+ *
+ *
+ */
+@Restricted(NoExternalUse.class)
+@Extension
+public class GeneratorController extends RunListener<Run> {
+    private boolean autostart = false;
+
+    // FIXME the accounting for tasks is broken, sometimes it overcounts, sometimes it undercounts.
+    ConcurrentHashMap<String, LoadGenerator> registeredGenerators = new ConcurrentHashMap<String, LoadGenerator>();
```

Comment:
favour package scoped getters rather than field access... all fields should be private